src/profile-logic/process-profile.js

```diff
@@ -835,14 +835,35 @@ export function processProfile(
  * Take a processed profile and remove any non-serializable classes such as the
  * StringTable class.
  */
-export function serializeProfile(profile: Profile): string {
+export function serializeProfile(
+  profile: Profile,
+  includeNetworkUrls: boolean = true
+): string {
   // stringTable -> stringArray
   const newProfile = Object.assign({}, profile, {
     threads: profile.threads.map(thread => {
-      const stringTable = thread.stringTable;
+      const stringArray = thread.stringTable.serializeToArray();
       const newThread = Object.assign({}, thread);
       delete newThread.stringTable;
-      newThread.stringArray = stringTable.serializeToArray();
+      if (includeNetworkUrls === false) {
+        for (let i = 0; i < newThread.markers.data.length; i++) {
```

Comment:
nit: you can use `newThread.markers.length`, see https://github.com/devtools-html/perf.html/blob/3d02369e79cb77640db96889b762074fedbfa240/src/types/profile.js#L71